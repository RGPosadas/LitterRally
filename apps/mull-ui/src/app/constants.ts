export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
