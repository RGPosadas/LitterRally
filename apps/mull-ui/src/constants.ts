import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
