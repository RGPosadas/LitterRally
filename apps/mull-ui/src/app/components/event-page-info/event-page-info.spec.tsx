import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import EventPageInfo from './event-page-info';
import { dummyEvent } from '../../../constants';

describe('EventPageInfo', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<EventPageInfo event={dummyEvent} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<EventPageInfo event={dummyEvent} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
