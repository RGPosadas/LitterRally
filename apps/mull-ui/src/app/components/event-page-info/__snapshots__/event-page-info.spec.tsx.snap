// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventPageInfo should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="info-row"
  >
    <img
      className="row-icon"
      src="https://i.pinimg.com/originals/0c/3b/3a/0c3b3adb1a7530892e55ef36d3be6cb8.png"
    />
    <p
      className="row-text"
    >
      Test user
    </p>
    <svg
      className="row-icon"
    >
      message.svg
    </svg>
  </div>
  <div
    className="info-row"
  >
    <svg
      className="row-icon"
    >
      location.svg
    </svg>
    <p
      className="row-text"
    >
      1260 Remembrance Road Montreal, Qc
    </p>
    <svg
      className="row-icon"
    >
      map.svg
    </svg>
  </div>
  <div
    className="info-row"
  >
    <svg
      className="row-icon"
    >
      description.svg
    </svg>
    <p
      className="row-text"
    >
      Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.
    </p>
  </div>
  <div
    className="info-row"
  >
    <svg
      className="row-icon"
    >
      event-restriction.svg
    </svg>
    <p
      className="row-text"
    >
      Everyone
    </p>
  </div>
</div>
`;
